core/course_controller/CourseController.py

```diff
@@ -105,7 +105,7 @@ def get_course(course_code: str, semester_name: str) -> CourseOverview:
         return CourseOverview(
             code=course.course_code,
             name=course.course_name,
-            semester=course.semester,
+            semester=course.semester.semester_name,
```

Comment:
It looks like there is a mistake for the types on course overview, the semester type should just be string. Can you change that in the static data classes file and make sure it doesn't cause issues anywhere?